python/pyspark/ml/torch/ssh_manager.py

```diff
@@ -0,0 +1,55 @@
+import os
+import subprocess
+import shutil
+from typing import (List)
+
+from pyspark.ml.torch.utils import write_to_location
+
+class SSHEnvManager:
+    """Is responsible for writing to the known_hosts file, setting up authorized users, and more"""
+    KNOWN_HOSTS = "/root/.ssh/known_hosts"
+    KNOWN_HOSTS_TEMP = "/root/.ssh/known_hosts_temp"
+
+    def __init__(self):
+        self.known_hosts_exists = os.path.exists(SSHEnvManager.KNOWN_HOSTS)
+        if self.known_hosts_exists:
+            shutil.copyfile(SSHEnvManager.KNOWN_HOSTS, SSHEnvManager.KNOWN_HOSTS_TEMP)
+
+    def create_ssh_key(self, ssh_key_path: str):
+        if not os.path.exists(ssh_key_path):
+            print(f"Creating the ssh key to {ssh_key_path}")
+            cmd_status = subprocess.run(["ssh-keygen", "-t", "rsa", "-f", ssh_key_path, "-q", "-N", ""])
```

Comment:
Do we need the trailing empty string? If so, leave a comment explaining why

Reply:
Also, capture_output=True here and include the output in the RuntimeError message

Reply:
I would recommend doing something like `f"ssh-keygen -t rsa -f {ssh_key_path} -q -N".split(" ")` for better changes.